File: problems/src/test/java/LongestPalindromicSubstringTest.java
import org.junit.Test;
import testing.Sample;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class LongestPalindromicSubstringTest extends SampledTest {
    @Override
    public String getFileName() {
        return "longest_palindromic_substring.json";
    }

    @Test
    public void test() {
        for (TypedSample sample : getSamples(TypedSample[].class)) {
            for (LongestPalindromicSubstring solution : LongestPalindromicSubstring.values()) {
                assertWithMessage(sample.getMessage())
                    .that(solution.longestPalindrome(sample.input.s))
                    .isEqualTo(sample.output);
            }
        }
    }

    public static class TypedSample extends Sample<TypedSample.Input, String> {
        public static class Input {
            public String s;
        }
    }
}

File: typescript/__tests__/problems1_100/longest-palindromic-substring.test.ts
import {strictEqual} from 'assert';
import SOLUTIONS from '../../src/problems1_100/longest-palindromic-substring';
import getSamples from '../../../testing/js/src/sample/samples';

type Sample = {
  input: {
    s: string,
  },
  output: string,
  message: string,
}

describe(
    'LongestPalindromicSubstring',
    () =>
        test(
            'test',
            () =>
                SOLUTIONS.forEach(solution =>
                    getSamples().forEach((sample: Sample) =>
                        strictEqual(
                            solution.longestPalindrome(sample.input.s),
                            sample.output,
                            sample.message,
                        ),
                    ),
                ),
        ),
);
